backend/api/auth.py

```
@@ -0,0 +1,297 @@
+    if not credentials.credentials.startswith("tk_"):
+        raise HTTPException(
+            status_code=status.HTTP_401_UNAUTHORIZED,
+            detail="Invalid API key format",
+        )
+    
+    # Hash the provided key to compare with stored hash
+    key_hash = hashlib.sha256(credentials.credentials.encode()).hexdigest()
+    
+    api_key = db.query(APIKey).filter(
+        APIKey.key_hash == key_hash,
+        APIKey.is_active == True
```

Ruff (0.12.2)

Avoid equality comparisons to `True`; use `APIKey.is_active:` for truth checks

Replace with `APIKey.is_active`

(E712)